wai-aria/role/contextual-roles.html

```diff
@@ -0,0 +1,102 @@
+<!doctype html>
+<html>
+<head>
+  <title>Contextual Role Verification Tests</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+  <script src="/wai-aria/scripts/aria-utils.js"></script>
+</head>
+<body>
+
+<p>Verifies Required Context Roles from <a href="https://www.w3.org/TR/wai-aria-1.2/#scope">5.2.7 Required Context Role</a></p>
+<p>Also verifies Required Owned Roles from <a href="https://www.w3.org/TR/wai-aria-1.2/#mustContain">5.2.6 Required Owned Role</a></p>
+
+<!-- Notes for "required context" testing:
+
+- See Computed Role for more details on role computation when an element lacks required context: 
+https://w3c.github.io/core-aam/#roleMappingComputedRole 
+
+- See ARIA 1.2 spec for full listing of "Required Context Role" for each element:
+https://www.w3.org/TR/wai-aria-1.2/
+
+Identified roles with "Required Context" excluding abstract roles (e.g., child -> parent):
+
+- caption -> {figure, grid, table, treegrid}
```

Comment:
I don't recall that we test `figure -> caption` anywhere. That seems like a good one for this file.

Reply:
The others may be in the grid-, table-, and tree-specific files, but maybe more variants are needed.

Reply:
@cookiecrook For your comment regarding `figure -> caption`, did you mean `caption -> figure`? I do have a test for a lone `role="caption"` within a `<p>` but want to ensure I didn't miss other related tests. Thank you!

Reply:
I should have said [`figure -> figcaption`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption) which I recalled (I was wrong) got a `caption` role according to HTML-AAM, [but it doesn't](https://w3c.github.io/html-aam/#el-figcaption), so no need to add another test here. 

Thread resolved.